cli/tools/bench.rs

```diff
@@ -123,6 +132,74 @@ pub trait BenchReporter {
   fn report_result(&mut self, desc: &BenchDescription, result: &BenchResult);
 }
 
+#[derive(Debug, Serialize)]
+struct JsonReporterResult {
+  runtime: String,
+  cpu: String,
+  origin: String,
+  group: Option<String>,
+  name: String,
+  baseline: bool,
+  result: BenchResult,
+}
+
+impl JsonReporterResult {
+  fn new(
+    origin: String,
+    group: Option<String>,
+    name: String,
+    baseline: bool,
+    result: BenchResult,
+  ) -> Self {
+    Self {
+      runtime: format!("{} {}", get_user_agent(), env!("TARGET")),
+      cpu: mitata::cpu::name(),
+      origin,
+      group,
+      name,
+      baseline,
+      result,
+    }
+  }
+}
+
+#[derive(Debug, Serialize)]
+struct JsonReporter(Vec<JsonReporterResult>);
+impl JsonReporter {
+  fn new() -> Self {
+    Self(vec![])
+  }
+}
+
+impl BenchReporter for JsonReporter {
+  fn report_group_summary(&mut self) {}
+  #[cold]
+  fn report_plan(&mut self, _plan: &BenchPlan) {}
+
+  fn report_end(&mut self, _report: &BenchReport) {
+    match write_json_to_stdout(self) {
+      Ok(_) => (),
+      Err(e) => println!("{e}"),
+    }
+  }
+
+  fn report_register(&mut self, _desc: &BenchDescription) {}
+
+  fn report_wait(&mut self, _desc: &BenchDescription) {}
+
+  fn report_output(&mut self, _output: &str) {}
+
+  fn report_result(&mut self, desc: &BenchDescription, result: &BenchResult) {
+    self.0.push(JsonReporterResult::new(
+      desc.origin.clone(),
```

Comment:
doc JSON output uses `location.filename` (together with `location.line` and `location.col`, and the error stack uses `fileName`. Maybe align with either or?